developer/tutorials/Architecture of nopCommerce.md

```diff
@@ -0,0 +1,71 @@
+# Architecture of nopCommerce
+
+## Introduction
+
+nopCommerce is a highly customizable and flexible, multi-store, multi-vendor, SEO friendly, full featured open-source E-Commerce solution. Which is build on top of Microsoft `ASP.NET Core` framework. nopCommerce is always up to date with latest technology and follows best practices.
+
+## Overview
+
+This document provides a comprehensive architectural overview of the nopCommerce system. This documentation is targeted for new nopCommerce developers. In this documentation we will be exploring each projects in nopCommerce solution, dependencies between those projects etc.
+
+## Overview of nopCommerce Architecture
+
+nopCommerce is one of the most popular and successful `Dot NET based` open-source `E-Commerce` solution. Success of nopCommerce is not only because it contains most of the features required by modern E-Commerce solution out of the box and it's UI is highly customizable and User friendly, but also because the nopCommerce solution is equally organized and developer friendly. The main strength of nopCommerce is its flexible, extendable architecture and well organized source code. The nopCommerce architecture is very close to the onion architecture. Which is mainly focused on controlling the code coupling. According to this architecture all code can depend on layers more central, but code cannot depend on layers further out from the core.  In other words, all coupling is toward the center.
+
+![nopCommerceArchitecturalDiagram](_static/Architecture-of-nopCommerce/nopCommerceArchitecture.png)
```

Comment:
The proposed diagram and method for describing the architecture are original. But the text traces the idea that there is a level of access to data, a level of presentation, and a level of business logic. I think it will not be out of place to add a schematic diagram in such a context. You can watch something similar in the video from [this article](https://medium.com/nopcommerce/the-architecture-of-nopcommerce-ecommerce-asp-net-core-based-open-source-platform-fb6b7fbc134c) to understand what I mean.